websockets/websockets.go

```diff
@@ -399,7 +429,15 @@ func (w *webSocket) loop() {
 						return
 					}
 					size := len(msg.data)
-					err := w.conn.WriteMessage(msg.mtype, msg.data)
+
+					err := func() error {
+						if msg.deadline.IsZero() {
+							return w.conn.WriteMessage(msg.mtype, msg.data)
+						}
+
+						// WriteControl is concurrently okay
+						return w.conn.WriteControl(msg.mtype, msg.data, msg.deadline)
```

Comment:
nit: Maybe just have a default `deadline` and instead the msg should have "isControl" boolean :thinking:

Reply:
Should be good with ae1e0e6